lib/backend/entity_info.dart

```diff
@@ -41,14 +40,18 @@ class EntityInfo {
   }
 
   @override
-  bool operator ==(Object? other) {
+  // TODO(@HrX03): is this correct?
```

Comment:
doesn't seem so? if the class can be made const usually it means it's immutable? i wonder what is going wrong here, maybe a false positive?

Reply:
The method '==' should not be overridden in classes not annotated with '@immutable'.
Try removing the override or annotating the class with '@immutable'.dart[avoid_equals_and_hash_code_on_mutable_classes](https://dart.dev/lints/avoid_equals_and_hash_code_on_mutable_classes)

when we add @ immutable its gone, is this okay for you?

Reply:
Sounds fine to me

Reply:
added immutable, alternatively I can remove

avoid_equals_and_hash_code_on_mutable_classes